src/azul/chalice.py

```diff
@@ -463,34 +467,34 @@ def __authenticate(self):
             log.info('Authenticated request as %r', auth)
 
     def _log_request(self):
-        if log.isEnabledFor(logging.INFO):
-            assert self.current_request is not None
-            context = self.current_request.context
-            request_info = {
-                'query': self.current_request.query_params,
-                'headers': self.current_request.headers
-            }
-            log.info('Received %s request for %r, with %s.',
-                     context['httpMethod'],
-                     context['path'],
-                     json.dumps(request_info, cls=self._LogJSONEncoder))
+        assert self.current_request is not None
+        context = self.current_request.context
+        request_info = {
+            'query': self.current_request.query_params,
+            'headers': self.current_request.headers
+        }
+        log.info('Received %s request for %r, with %s.',
+                 context['httpMethod'],
+                 context['path'],
+                 json.dumps(request_info, cls=self._LogJSONEncoder))
+        self._log_body('request', self.current_request.json_body)
 
     def _log_response(self, response):
-        if log.isEnabledFor(logging.DEBUG):
-            n = 1024
-            if isinstance(response.body, str):
-                body = response.body[:n]
-            else:
-                body = json_head(n, response.body)
-            log.debug('Returning %i response with headers %s. '
-                      'See next line for the first %i characters of the body.\n%s',
-                      response.status_code,
-                      json.dumps(response.headers, cls=self._LogJSONEncoder),
-                      n, body)
+        log.info('Returning %i response with headers %s.',
+                 response.status_code, json.dumps(response.headers, cls=self._LogJSONEncoder))
+        self._log_body('response', response.body)
+
+    def _log_body(self, body_type: BodyType, body: str | JSON | None):
+        if body is None:
+            log.info(http_body_log_message(body_type, None))
         else:
-            log.info('Returning %i response. '
-                     'To log headers and body, set AZUL_DEBUG to 1.',
-                     response.status_code)
+            if isinstance(body, Mapping):
+                body = json.dumps(body)
+            if log.isEnabledFor(logging.DEBUG):
+                message = http_body_log_message(body_type, body, verbatim=config.debug > 1)
+                log.debug(message)
+            else:
+                log.info('… with %s body size of %i bytes', body_type, len(body))
```

Comment:
If we need to, we can get the body size from the API Gateway logs, so removing it would enable us to eliminate the relatively expensive `json.dumps` above.